envoyauth/response.go

```diff
@@ -0,0 +1,292 @@
+package envoyauth
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	ext_core_v3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	ext_type_v3 "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/open-policy-agent/opa-envoy-plugin/internal/util"
+	"github.com/open-policy-agent/opa/metrics"
+)
+
+//EvalResult - Captures the result from evaluating a query against an input
+type EvalResult struct {
+	Revision   string // Deprecated: Use `revisions` instead.
+	Revisions  map[string]string
+	DecisionID string
+	TxnID      uint64
+	Decision   interface{}
+	Metrics    metrics.Metrics
+}
+
+//StopFunc should be called as soon as the evaluation is finished
+type StopFunc = func()
+
+//NewEvalResult creates a new EvalResult and and a StopFunc that is used to stop the timer for metrics
```

Comment:
That is really your call. It would essentially mean that users of this method must know that they have to start + stop the timer on the metrics object so that the results show up in the decision log to my understanding. It was code I had to copy in my use of this codebase. The way it is now, I was forced to think about calling the stop function.